contrib/vagrant/provision-util.sh

```diff
@@ -190,7 +190,8 @@ os::provision::get-network-plugin() {
   local default_plugin="${subnet_plugin}"
 
   if [[ "${plugin}" != "${subnet_plugin}" &&
-          "${plugin}" != "${multitenant_plugin}" ]]; then
+          "${plugin}" != "${multitenant_plugin}" &&
+          "${plugin}" != "${cni}" ]]; then
```

Comment:
Can we do what @marun suggested, if no plugin is specified then use the default plugin otherwise don't do any validation and use the given plugin? With that, we don't need to add these checks for any other plugins.
Also squash your commits.